woocommerce/woocommerce
# Fix the bug where CES requires a double-click on Give feedback

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:






Closes #52515 

In https://github.com/woocommerce/woocommerce/pull/43164, we added a logic to update `woocommerce_ces_shown_for_actions` option when a notice is shown. This caused a bug where `CustomerEffortScoreTracks` re-renders `CustomerEffortScore`, resulting in a new notice.

This PR replaces `updateOption` call with `apiFetch` to avoid re-render. 

We could try refactoring the code to fix this issue, but I didn't want to refactor code that has been working fine for a couple of years when a simple fix is available. 




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site using this branch.
2. Complete the core profiler (you can skip the extensions page).
3. Navigate to `Products -> Add new` and fill in the product form.
4. Publish the product.
5. A snackbar should appear; do not click on it.
6. Navigate to a different page and then return to `Products -> Add new` page.
7. Publish another product.
8. Confirm that the snackbar does not appear this time.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
A fix a bug where users need to click Give feedback twice.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
